### PR TITLE
[soem] Add new port

### PR DESCRIPTION
**Pull request by seanyen**
Add a new port `SOEM` - https://github.com/OpenEtherCATsociety/SOEM.

### CODE REVIEW

File: ports/soem/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,36 @@
+include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/soem/portfile.cmake]: `include(vcpkg_common_functions)` is no longer needed.
Could you please remove this line?

File: ports/soem/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE
+    ${CURRENT_PACKAGES_DIR}/bin
+    ${CURRENT_PACKAGES_DIR}/debug/bin
+    ${CURRENT_PACKAGES_DIR}/debug/include)
+
+# Handle copyright
+file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/soem)
+file(RENAME ${CURRENT_PACKAGES_DIR}/share/soem/LICENSE ${CURRENT_PACKAGES_DIR}/share/soem/copyright)
```
Comment (NancyLi1013) [inline ports/soem/portfile.cmake]: Could you update this
```
file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/soem)
file(RENAME ${CURRENT_PACKAGES_DIR}/share/soem/LICENSE ${CURRENT_PACKAGES_DIR}/share/soem/copyright)
```
as `file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright) `?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by seanyen** [comment]
@NancyLi1013 Thanks for the feedback. I have made the respective changes.

**Comment by NancyLi1013** [comment]
Hi @seanyen thanks for this PR.
I noticed that this port failed on` x64-uwp` and `arm-uwp` triplets from CI system.
Are these expected? If not, please try to fix them.

If this port doesn't support arm, please add `vcpkg_fail_port_install(ON_TARGET "UWP")` to the top of `portfile.cmake`.
Also update [baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) with the following content:
```
soem:x64-uwp=fail
soem:arm-uwp=fail
```

File: ports/soem/CONTROL [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Source: soem
+Version: 1.4.0-1
```
Comment (PhoebeHui) [inline ports/soem/CONTROL]: The revision should be 1.4.0

File: ports/soem/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    SHA512 4f1118e48908552c664f45f369e6fde09aba11ce72fa6a3a15f5805a280d77d998e0c92f1647185981aebc717037f62bac8b362444317405e795988a2ddbddd3
+)
+
+vcpkg_extract_source_archive_ex(
+    OUT_SOURCE_PATH SOURCE_PATH
+    ARCHIVE ${ARCHIVE}
+    REF ${SOEM_VERSION}
+    PATCHES
+        winpcap.patch
+)
```
Comment (PhoebeHui) [inline ports/soem/portfile.cmake]: You can use follow code instead.
```
  vcpkg_from_github(
        OUT_SOURCE_PATH SOURCE_PATH
        REPO OpenEtherCATsociety/SOEM
        REF abbf0d42e38d6cfbaa4c1e9e8e07ace651c386fd #v1.4.0
        SHA512 2967775c6746bb63becea5eb12f136c184bbf874e1e5e8753374bfc212ec9cefbf1159350e79627b978af3562d261b61c50f38936a425c4d9c70598a1d136817
        HEAD_REF master
        PATCHES 
             winpcap.patch
    )
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

**Comment by seanyen** [comment]
@NancyLi1013 @PhoebeHui Thanks for the feedback. I have revised this PR respectively.

**Review by PhoebeHui — APPROVED** [review]

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**